docker/Dockerfile.isce3.dev

```diff
@@ -0,0 +1,84 @@
+# A Dockerfile to install ISCE3 from source code,
+# so that the up-to-date features can be tested.
+#
+# Instruction
+#
+
+FROM oraclelinux:8
+
+LABEL author="OPERA ADT" \
+    description="s1 cslc 0.2 Gamma release" \
+    version="0.2-gamma"
+
+RUN yum -y update &&\
+    yum -y install curl git make unzip &&\
+    adduser compass_user &&\
+    mkdir -p /.config && chmod -R 777 /.config &&\
+    mkdir -p /.cache && chmod -R 777 /.cache
+
+USER compass_user
+
+RUN mkdir -p /home/compass_user/OPERA &&\
+    chmod -R 755 /home/compass_user
+
+ENV CONDA_PREFIX=/home/compass_user/miniconda3
+
+# install Miniconda
+WORKDIR /home/compass_user
+RUN curl -sSL https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -o miniconda.sh &&\
+    bash miniconda.sh -b -p ${CONDA_PREFIX} &&\
+    rm $HOME/miniconda.sh
+
+ENV PATH=${CONDA_PREFIX}/bin:${PATH}
+RUN ${CONDA_PREFIX}/bin/conda init bash
+
+# copy COMPASS and set compass_user as owner
+COPY --chown=compass_user:compass_user . /home/compass_user/OPERA/COMPASS
+
+# create CONDA environment
+RUN conda create --name "COMPASS" --file /home/compass_user/OPERA/COMPASS/docker/specfile.isce3.dev.txt
+
+SHELL ["conda", "run", "-n", "COMPASS", "/bin/bash", "-c"]
+
+# Install ISCE3
+RUN mkdir -p /home/compass_user/OPERA/ISCE3/install &&\
+    mkdir -p /home/compass_user/OPERA/ISCE3/build &&\
+    cd /home/compass_user/OPERA/ISCE3/build &&\
+    cmake -DCMAKE_INSTALL_PREFIX=/home/compass_user/OPERA/ISCE3/install -DWITH_CUDA=OFF /home/compass_user/OPERA/COMPASS/isce-develop &&\
+    make -j32 VERBOSE=ON &&\
+    make install &&\
+    rm -rf /home/compass_user/OPERA/ISCE3/build/* &&\
+    cd $CONDA_PREFIX/lib &&\
+    find /home/compass_user/OPERA/ISCE3/install/lib64 -name "*.so*" -exec ln -s {} \; &&\
+    cd `find $CONDA_PREFIX -name "site-packages"|sort|tail -1` &&\
+    find /home/compass_user/OPERA/ISCE3/install/packages -maxdepth 1 -type d|sort|awk 'NR>1'|xargs -I{} ln -s {} &&\
+    rm -rf /home/compass_user/OPERA/ISCE3/build
+
+WORKDIR /home/compass_user/OPERA
+# Install ISCE3
+
+
+# installing OPERA s1-reader
+RUN curl -sSL https://github.com/opera-adt/s1-reader/archive/refs/tags/v0.1.7.tar.gz -o s1_reader_src.tar.gz &&\
+    tar -xvf s1_reader_src.tar.gz &&\
+    ln -s s1-reader-0.1.7 s1-reader &&\
+    rm s1_reader_src.tar.gz &&\
+    python -m pip install ./s1-reader
+
+# installing RAiDER
+RUN mkdir RAiDER &&\
+    git clone https://github.com/dbekaert/RAiDER.git RAiDER &&\
+    cd RAiDER &&\
+    git checkout tags/v0.4.2 &&\
+    cd .. &&\
+    python -m pip install ./RAiDER
+
+
+
```

Comment:
```suggestion
```
Extra space clean up

Reply:
Closing this comment as I've take out the on-the-fly ISCE3 builder. I'll take another look at the Dockerfile based on condo-based ISCE3.